### ISSUE TITLE
Multiple crashes when checking broken files

### ISSUE THREAD

**Issue by qarmin** [quick-lint/quick-lint-js#974]
quick-lint-js version 2.12.0

command
```
quick-lint-js app.js
```
crashes in several ways
```
/__w/quick-lint-js/quick-lint-js/dist/debian/build/quick-lint-js-2.12.0/src/quick-lint-js/fe/variable-analyzer.cpp:939: internal check failed in report_error_if_variable_declaration_conflicts: false

/__w/quick-lint-js/quick-lint-js/dist/debian/build/quick-lint-js-2.12.0/src/quick-lint-js/fe/parse-statement.cpp:2236: internal check failed in parse_and_visit_typescript_declare_namespace: parsed_statement

/__w/quick-lint-js/quick-lint-js/dist/debian/build/quick-lint-js-2.12.0/src/quick-lint-js/fe/variable-analyzer.cpp:807: internal check failed in report_error_if_assignment_is_illegal: false
```
with this files - [Broken_81.zip](https://github.com/quick-lint/quick-lint-js/files/11183403/Broken_81.zip)

**Comment by strager** [comment]
Thanks for the reports!

**Comment by strager** [comment]
I fixed the last of the issues in commit 449a12c8f0ce254f5fafb9d6641aa98290a3463a. Thanks again for reporting.
